lib/dap/filter/ldap.rb

```diff
@@ -0,0 +1,53 @@
+module Dap
+module Filter
+
+require 'openssl'
+
+require 'dap/proto/ldap'
+
+#
+# Decode an LDAP SearchRequest probe response
+#
+class FilterDecodeLdapSearchResult
+  include BaseDecoder
+
+  #
+  # Decode an LDAP SearchRequest probe response
+  #
+  # @param data [String] Binary string containing raw response from server
+  # @return [Hash] Hash containing all LDAP responses
+  #
+  def decode(data)
+    return unless data.length > 2
+
+    # RFC 4511 - 4.5.2 SearchResult contains zero or more SearchResultEntry or
+    # SearchResultReference messages followed by a single SearchResultDone
+    # message.  OpenSSL::ASN1.decode doesn't handle the back to back Sequences
+    # well, so identify the lengths and split them into individual ASN1 elements
+    messages = Dap::Proto::LDAP.split_messages(data)
+    return unless messages
```

Comment:
`messages` could be empty or `nil` here, depending on where `split_messages` returns.  Thats fine if that is the way `split_messages` is supposed to work, but just update the comments accordingly.  Personally, I'd update so it always returns a possibly empty array.